roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -38,6 +41,31 @@ impl JobDeclaratorDownstream {
     }
 }
 
+pub fn clear_declared_mining_job(
```

Comment:
as suggested in this [comment](https://github.com/stratum-mining/stratum/pull/868#issuecomment-2226012537), this function cannot clear from jds mempool transactions declared in old jobs.
I tested the following function and seems to work fine, even though requires more testing for other devs. Also, there are some suggestions left as a comment.
```rust
pub fn clear_declared_mining_job(
    mining_job: DeclareMiningJob,
    mempool: Arc<Mutex<JDsMempool>>,
) -> Result<(), Error> {
    // If there is an old declared mining job, remove its transactions from the mempool
    // Retrieve necessary data from the old job
    let transactions_to_remove = mining_job.tx_short_hash_list.inner_as_ref();
    if transactions_to_remove.is_empty() {
        debug!("No transactions to remove from mempool");
        return Ok(()); };
    let nonce = mining_job.tx_short_hash_nonce;

    for short_id in transactions_to_remove {
        // try to return en error if the following lock results in an error 
        let _ = mempool.safe_lock(|mempool_| {
            // try to manage this unwrap, use .ok_or() method to return the proper error
            match mempool_.to_short_ids(nonce).unwrap().get(short_id){
               Some(transaction_with_hash) => {
                   let txid = transaction_with_hash.id;
                   match mempool_.mempool.remove(&txid) {
                       Some(transaction) => {
                           debug!("Fat transaction {:?} in job with request id {:?} removed from mempool", transaction, mining_job.request_id);
                           info!("Fat transaction {:?} in job with request id {:?} removed from mempool", txid, mining_job.request_id);
                       },
                       None => info!("Thin transaction {:?} in job with request id {:?} removed from mempool", txid, mining_job.request_id)
                   }
               },
               None => debug!("Transaction with short id {:?} not found in mempool while clearing old jobs", short_id),
            }
        });
    };
    Ok(())
}
```